### PR TITLE
Add studio support

### PR DESCRIPTION
**Pull request by bluepilledgreat**
#36 
#516 

TODO checklist:
- [x] RegisterProgramSize
- [x] Update "RobloxPlayerBeta" references

### CODE REVIEW

**Review by EasternBloxxer — APPROVED** [review]
LGTM

File: Bloxstrap/UI/ViewModels/Bootstrapper/ByfronDialogViewModel.cs [hunk 1]
```diff
@@ -20,7 +22,10 @@ public string VersionText
         {
```
Comment (nakoyasha) [inline Bloxstrap/UI/ViewModels/Bootstrapper/ByfronDialogViewModel.cs:22]: I think it'd be better to pass version info (ex. guid, string, etc) as a constructor argument. Far cleaner and avoids having to figure out it out here.

**Review by nakoyasha — CHANGES_REQUESTED** [review]

**Comment by nakoyasha** [comment]
Overall LGTM, but I feel like Studio and Player should be separated into their own "instances", I don't think it's the greatest idea to have both share the same settings, but that's out of the scope  of this PR.

**Comment by EasternBloxxer** [comment]
> Overall LGTM, but I feel like Studio and Player should be separated into their own "instances", I don't think it's the greatest idea to have both share the same settings, but that's out of the scope  for this PR.

Yeah as there are flags for player that make studio not launch

**Comment by passionfr7it** [comment]
i support !

File: Bloxstrap/App.xaml.cs [hunk 2]
```diff
@@ -227,6 +237,25 @@ protected override void OnStartup(StartupEventArgs e)
...
+                    commandLine = HttpUtility.UrlDecode(LaunchArgs[0]);
+                    isStudioLaunch = true;
+                    isStudioAuth = true;
+                }
+                else if (LaunchArgs[0] == "-ide")
+                {
+                    isStudioLaunch = true;
+                    if (LaunchArgs.Length >= 2)
+                        commandLine = $"-task EditFile -localPlaceFile \"{LaunchArgs[1]}\"";
+                }
```
Comment (apprehensions) [inline Bloxstrap/App.xaml.cs:258]: You can simply pass Studio `-protocolString`, the same can be said for player by simply providing the launch URI, as player and studio can handle the URI independently!

**Comment by pizzaboxer** [comment]
Looks good, provids a great base to work with.

**Pull request merged**